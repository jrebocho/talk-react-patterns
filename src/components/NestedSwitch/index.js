import React, { Component } from 'react'
import './NestedSwitch.css'
import StatelessSwitch from '../StatelessSwitch'

class NestedSwitch extends Component {
  constructor(props) {
    super(props)
    this.state = { on: false }
    this.toggle = this.toggle.bind(this)
  }

  toggle() {
    this.setState(prevState => ({
      on: !prevState.on
    }))
  }

  render() {
    const boxColor = this.state.on ? 'green' : 'red'

    return (
      <div className={`outer-box ${boxColor}`} onClick={this.toggle}>
        <YellowBox on={this.state.on} />
      </div>
    )
  }
}

const YellowBox = ({ on }) => (
  <div className="yellow">
    <StatelessSwitch on={on} />
  </div>
)

export default NestedSwitch
