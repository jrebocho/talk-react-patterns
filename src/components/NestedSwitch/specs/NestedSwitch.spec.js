import React from 'react'
import { shallow } from 'enzyme'
import NestedSwitch from '../../NestedSwitch'

describe('<NestedSwitch />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<NestedSwitch />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('when the component is clicked', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<NestedSwitch />)
      wrapper.simulate('click')

      expect(wrapper).toMatchSnapshot()
    })
  })
})
