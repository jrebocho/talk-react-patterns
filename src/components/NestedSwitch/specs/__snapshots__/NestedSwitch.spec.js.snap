// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NestedSwitch /> renders correctly 1`] = `
<div
  className="outer-box red"
  onClick={[Function]}
>
  <YellowBox
    on={false}
  />
</div>
`;

exports[`<NestedSwitch /> when the component is clicked renders correctly 1`] = `
<div
  className="outer-box green"
  onClick={[Function]}
>
  <YellowBox
    on={true}
  />
</div>
`;
